src/genophenocorr/preprocessing/_uniprot.py

```diff
@@ -64,8 +68,8 @@ def annotate(self, protein_id: str) -> typing.Sequence[ProteinMetadata]:
                     self._logger.warning(f"No features for {protein_id}")
                 protein_list.append(ProteinMetadata(protein_id, protein_name, all_features_list))
             else:
-                self._logger.warning(f"ID {protein_id} did not match")
-        self._logger.warning(f'Protein ID {protein_id} got {len(protein_list)} results')
-
+                self._logger.warning(f"UniProt did not return a protein ID that matches the ID we searched for: {protein_id} not in {unis}")
+        if len(protein_list) > 1:
```

Comment:
People are usually interested in `logging.INFO` messages. However, here we'll produce an INFO message for each successful Uniprot query, which will, hopefully, be >99% of time. It would be better to make this a debug statement, or just not log anything.

Reply:
` if len(protein_list) > 1:
     self._logger.info(f'UniProt found {len(protein_list)} results for ID {protein_id}')`

Think this is the line you are referring to, and this will actually rarely run since I have almost never gotten more than one UniProt result for a given ID.

Reply:
OK, let's leave it here if it occurs rarely.